### PR TITLE
[Profiler] Adjust profiler tests for .NET 8

### PR DESCRIPTION
**Pull request by gleocadie**
## Summary of changes

Adjust checks for profiler tests for .NET 8

## Reason for change

really ?
## Implementation details

Adjust the checks according to the CLR version.

## Test coverage

The current tests.

### CODE REVIEW

File: profiler/test/Datadog.Profiler.IntegrationTests/Exceptions/ExceptionsTest.cs [hunk 1]
```diff
@@ -46,16 +46,11 @@ public void ThrowExceptionsInParallel(string appName, string framework, string a
                     new StackFrame("|lm:Samples.ExceptionGenerator |ns:Samples.ExceptionGenerator |ct:ParallelExceptionsScenario |cg: |fn:ThrowExceptions |fg: |sg:(object state)"),
                     new StackFrame("|lm:System.Private.CoreLib |ns:System.Threading |ct:Thread |cg: |fn:StartCallback |fg: |sg:()"));
             }
-            else if (framework == "net7.0")
+            else if (framework == "net7.0" || framework == "net8.0")
```
Comment (andrewlock) [inline profiler/test/Datadog.Profiler.IntegrationTests/Exceptions/ExceptionsTest.cs:49]: Ok, I probably should have just tried that 😂 🤦‍♂️

**Review by andrewlock — APPROVED** [review]
Thanks!

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (4908)](https://github.com/DataDog/dd-trace-dotnet/tree/a7dcc8be9fd1133794139d7823b28f2e112bc07a)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/1d1be6175b9bd60050789327c53b4de5c264dba9)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4908) - mean (71ms)  : 62, 79
     .   : milestone, 71,
    master - mean (69ms)  : 64, 75
     .   : milestone, 69,

    section CallTarget+Inlining+NGEN
    This PR (4908) - mean (1,002ms)  : 986, 1019
     .   : milestone, 1002,
    master - mean (1,001ms)  : 983, 1018
     .   : milestone, 1001,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4908) - mean (106ms)  : 103, 109
     .   : milestone, 106,
    master - mean (106ms)  : 102, 109
     .   : milestone, 106,

    section CallTarget+Inlining+NGEN
    This PR (4908) - mean (688ms)  : 676, 701
     .   : milestone, 688,
    master - mean (683ms)  : 662, 703
     .   : milestone, 683,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4908) - mean (90ms)  : 87, 93
     .   : milestone, 90,
    master - mean (89ms)  : 85, 93
     .   : milestone, 89,

    section CallTarget+Inlining+NGEN
    This PR (4908) - mean (657ms)  : 635, 680
     .   : milestone, 657,
    master - mean (645ms)  : 626, 663
     .   : milestone, 645,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4908) - mean (188ms)  : 185, 192
     .   : milestone, 188,
    master - mean (188ms)  : 185, 191
     .   : milestone, 188,

    section CallTarget+Inlining+NGEN
    This PR (4908) - mean (1,133ms)  : 1114, 1151
     .   : milestone, 1133,
    master - mean (1,131ms)  : 1107, 1154
     .   : milestone, 1131,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4908) - mean (271ms)  : 267, 276
     .   : milestone, 271,
    master - mean (271ms)  : 267, 275
     .   : milestone, 271,

    section CallTarget+Inlining+NGEN
    This PR (4908) - mean (1,090ms)  : 1066, 1115
     .   : milestone, 1090,
    master - mean (1,093ms)  : 1073, 1112
     .   : milestone, 1093,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4908) - mean (261ms)  : 258, 264
     .   : milestone, 261,
    master - mean (261ms)  : 257, 264
     .   : milestone, 261,

    section CallTarget+Inlining+NGEN
    This PR (4908) - mean (1,055ms)  : 1031, 1078
     .   : milestone, 1055,
    master - mean (1,052ms)  : 1027, 1077
     .   : milestone, 1052,

```

**Review by chrisnas — APPROVED** [review]
LGTM

**Pull request merged**